community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -268,21 +303,14 @@ synchronized void releaseReadLock( Object tx ) throws LockNotFoundException
                         {
                             // found a write lock with all read locks
                             listItr.remove();
-                            if ( !we.element.movedOn )
-                            {
-                                we.waitingThread.interrupt();
-                                // ----
-                                break;
-                            }
+                            we.waitingThread.interrupt();
```

Comment:
I am a bit uncertain what is going on here. Above, setting movedOn to true is replaced with incrementing requests so i would expect the correspondence here to be if isNoRequests() to match movedOn check. What has changed?